### ISSUE TITLE
When an AnalogStick is used as an I2DInput, deadzone should be applied.

### ISSUE THREAD

**Issue by vchelaru** [vchelaru/FlatRedBall#841]
Currently deadzone is stored and applied at the Xbox360GamePad level. This is problematic because if an AnalogStick is directly used, the deadzone value does not apply. This is most obvious when using the Top Down behavior which relies on I2DInput for movement.

To fix this:

1. Deadzone values should be stored at the analog stick level
2. Analog stick `get` properties should always return values after considering deadzone, including any interface fulfilling properties
3. The Xbox360GamePad should fall back to the left stick when getting, and should apply both sticks when setting. If there are no sticks (these can be null for controllers) then return its own stored value.

In other words, using Xbox360GamePad's properties should still function exactly how it worked before, but now users can override the overall value by directly assigning deadzone values on individual controllers. It is possible to have the Xbox360GamePad deadzone values disagree with individual stick values, but that's up to the user to decide whether to rely on the overall values or individual stick values.

**Comment by vchelaru** [comment]
Fixed

b3b96fa1a5ed9f48a73f5f0a38fbc23e663afb3a
